google/auth/credentials.py

```diff
@@ -236,6 +223,40 @@ def with_non_blocking_refresh(self):
         self._use_non_blocking_refresh = True
 
 
+class CredentialsWithTrustBoundary(Credentials):
+    """Abstract base for credentials supporting trust boundary factory"""
+
+    def apply(self, headers, token=None):
+        super().apply(headers, token)
+        """Trust boundary value will be a cached value from global lookup.
```

Comment:
nit: Should this docstring be above the super call? Also, can the docstring describe what the function is doing, it's a little confusing currently